mila_datamodules/cli/prepare_torchvision.py

```diff
@@ -146,56 +256,286 @@ def __call__(
         return super()(root, *constructor_args, **constructor_kwargs)
 
 
+class Compose(AbstractPrepareVisionDataset[VD, P]):
+    def __init__(self, callables: list[AbstractPrepareVisionDataset]) -> None:
+        self.callables = callables
+
+    @runs_on_local_main_process_first
+    def __call__(
+        self,
+        root: str | Path = SLURM_TMPDIR / "datasets",
+        *dataset_args: P.args,
+        **dataset_kwargs: P.kwargs,
+    ) -> str:
+        for c in self.callables:
+            # TODO: Check that nesting `runs_on_local_main_process_first` decorators isn't a problem.
+            root = c(root, *dataset_args, **dataset_kwargs)
+        return root
+
+
 # NOTE: For some datasets, we have datasets stored in folders with the same structure. This here is
 # only really used to prevent repeating a bit of code in the dictionary below.
 # TODO: Find an exception to this rule and design this dict with that in mind.
-standardized_torchvision_dataset_dirs = {
+standardized_torchvision_datasets_dirs = {
     Cluster.Mila: "/network/datasets",
-    Cluster.Beluga: "/project/rpp-bengioy/data/curated",
+    Cluster.Beluga: "~/project/rpp-bengioy/data/curated",
```

Comment:
I'd avoid going through the $HOME filesystem if possible.

Reply:
Indeed we can do what's suggested below